firebase_admin/db.py

```diff
@@ -253,16 +267,13 @@ def transaction(self, transaction_update):
 
         tries = 0
         etag, data = self._get_with_etag()
-        val = transaction_update(data)
         while tries < _TRANSACTION_MAX_RETRIES:
-            success, etag, snapshot = self._update_with_etag(val, etag)
+            new_data = transaction_update(data)
+            success, etag, data = self._update_with_etag(new_data, etag)
```

Comment:
Yes, in the next iteration of the loop `etag` gets passed to the `_update_with_etag()` call.